PIPELINE-392 - Add files for GDAL 3.0.4

CJohnsonLehi opened the pull request:
The changes I made to rebrand this git repo for the lambda layers we use.

The requirements files have been paired down since we really don't need so much stuff in there. These just have what we need for our lambda function to execute properly.

Commit:
Add files for GDAL 3.0.4

gwfraley commented:
> The requirements files have been paired down since we really don't need so much stuff in there.

@CJohnsonLehi  - we actually use rasterio and several of these dependencies in other microservices. Suggest adding some optionality versus assuming what specific services need.

Building on the above comment. You could have just added the 3 extra dependencies in the requirements.txt of your lambda service without changing the layer at all. However, if you really don't need the 12 other dependencies, then that suggests we have at least 2 layers that could be interesting:
- basic gdal w/ python bindings
- rasterio focused

I'd suggest adding new requirements.txt files for the Vexcel use-cases / layers. And, leave the upstream requirements.txt files as-is.

CJohnsonLehi commented:
So the previous requirements file was called gdal-xxxx but it didn't actually contain gdal python packages. So I think I will create a new requirements file that contains the old, other stuff but give it a new name so as not to cause confusion and to clarify better what is in it.
But in reality the only parameter this docker image provides is the GDAL version. If we want to provide and extra parameter that allows us to pass in another requirements file, it will require reworking of the docker image and the make file. And therefore could be left to someone else to configure the next time there is a need for that. When will that be?

I'm not sure if that task is inside the scope of this task.

Commit:
Add two new requirement files

The python packages listed in these two files are what originally
in the requirement-gdalX.X.txt files.
I'm putting them in their own file so as to not lose the original
contents.
It will be up to the next person who actually has need of them, and will
use them to actually modify the docker image and the make file to
consume these: Since it appears these are actually unused at this time.

Commit:
Shorten the zip file name

It currently has just gdal in it. No need to make the zip file
extravagent and confusing with the name 'osgeo'.

Commit:
Rename files to rasterio

Commit:
Add better parameter for the requirements file

This adds a docker argument that better indicates which
requirements file to use. The version is still required, but
now we have greater flexibility for the requirements file
which a future change to the make file can benefit from.

gwfraley commented:
> So the previous requirements file was called gdal-xxxx but it didn't actually contain gdal python packages.

rasterio is built upon GDAL, and is an alternative to the GDAL python bindings

> So I think I will create a new requirements file that contains the old, other stuff but give it a new name so as not to cause confusion and to clarify better what is in it.

I would suggest not renaming the requirements files from the upstream repo, so that we can keep consistent with upstream and pull changes easily later.

> If we want to provide and extra parameter that allows us to pass in another requirements file

Seems reasonable within the scope of this PR

gwfraley commented on requirements-gdal2.4.txt:
```diff
@@ -1,12 +1,3 @@
\ No newline at end of file
+gdal==2.4
+packaging
+joblib
```
Seems like `packaging` and `joblib` should be added in the requirements file of the lambda that builds upon this layer, as opposed to including them in the layer directly.

Do you foresee needing these in _all_ services that make use of this lambda layer?

CJohnsonLehi replied:
No, in fact today in the lambda function I created this for, I removed the need for packaging and joblib. So I think I will delete those.

CJohnsonLehi commented:
> I would suggest not renaming the requirements files from the upstream repo, so that we can keep consistent with upstream and pull changes easily later.

Good point! I will make that change.

Commit:
Code review feedback

Leaving the original requirements files unchanged now.
This will make it easier for us to incorporate upstream
changes in the future if we need them.

sshort24 commented on Dockerfile:
```diff
@@ -2,11 +2,10 @@ ARG GDAL_VERSION
-RUN pip install cython==0.28
-
-COPY requirements-gdal${GDAL_VERSION}.txt requirements.txt
+COPY requirements-${REQ_VERSION}.txt requirements.txt
```
Now that we know the underlying root issue was inversion of 2.4 lat/lon parameters to lon/lat in 3.0, why do we need the extra complexity to handle both versions? Why not just commit to 3.0

CJohnsonLehi replied:
That is fine with me. But when you wrote to abandon the PR, what do you mean? Make the change in some other branch? (In this repo?)

sshort24 replied:
I now understand that the original cogeo repo supports both 2.4 and 2.5. This PR removes changes WE made to simplify merging future changes made to the original cogeo repo.

CJohnsonLehi replied:
2.4 and 3.0. I could make a second docker file, and a second makefile just for our changes... That would really make it easier to incorporate upstream changes.

sshort24 reviewed (changes requested):
I recommend we abandon this PR and standardize on GDAL v3.0

Commit:
Roll back changes to original files.

Add on files just for our changes that we need
to make a gdal only lambda layer (i.e. zip file).
Also dropped support for 2.4

Doing it this way will really make it easy to incorporate upstream
changes into this git repo.
